### ISSUE TITLE
Session input cursor should wait in column 0 during processing

### ISSUE THREAD

**Issue by JohnScholes** [Dyalog/ride#382]
### Describe the issue you are having
After entering a time-consuming expression, the input cursor is placed after six blanks in the following line. This makes it appear as if the result of the expression is null or a 0-row matrix.

### How do you reproduce the issue?
```
      ⊢⎕dl 5   ⍝ input cursor appears indented in the following line
```
11425291 send ["Execute",{"trace":0,"text":"      ⊢⎕dl 5\n"}]
11425301 recv ["EchoInput",{"input":"      ⊢⎕dl 5\n"}]
11425313 recv ["SetPromptType",{"type":0}]

**Comment by JohnScholes** [comment]
Alternatively, Ride could mimic ODE by having the input cursor _disappear_ during processing.

**Comment by abrudz** [comment]
This issue it made even worse by the lack of the IDE's status bar which indicates in red when `⎕SI` is non-empty.

**Comment by e9gille** [comment]
This is annoying. Monaco's way to indicate readOnly is to make the cursor solid. They have reserved the hidden state for internal use. I'll work around this.

@abrudz Status bar is in the pipeline.

**Comment by jayfoad** [comment]
This is nitpicking, but... what I see now is that after I type `⎕dl 2<ER>`, the cursor briefly appears in column 6 of the next line, and _then_ disappears for 2 seconds. Personally I think it would be nicer if we could leave the cursor visible in column 0 while the interpreter is busy (we don't have to mimic every aspect of the ODE's behaviour).

**Comment by JohnScholes** [comment]
Agreed: leaving the cursor visible in column 0 during processing would be nice.
